src/pybind/mgr/cephadm/inventory.py

```diff
@@ -122,6 +122,60 @@ def clear_queued_daemons(self) -> None:
         self._daemons = {}
 
 
+class DaemonRemovalQueue:
```

Comment:
You cover this in the commit message but the duplication is clear and I worry that in later will be never.